gnovm/cmd/gno/util.go

```diff
@@ -100,6 +100,64 @@ func gnoPackagesFromArgs(args []string) ([]string, error) {
 	return paths, nil
 }
 
+func gnoPackagePathsFromPattern(patterns []string) ([]string, error) {
+	paths := []string{}
+	recursiveLookup := false
+	for _, p := range patterns {
+		// TODO:
+		//     - Support more patterns
+		//     - Move this logic to separate helper function?
+		if strings.HasSuffix(p, "/...") {
+			recursiveLookup = true
+			p = strings.TrimSuffix(p, "/...")
+		}
+
+		info, err := os.Stat(p)
+		if err != nil {
+			return nil, fmt.Errorf("invalid file or package path: %w", err)
+		}
+
+		// if the passed arg is a file, then we'll just add it to the list
+		if !info.IsDir() {
+			paths = append(paths, p)
+			continue
+		}
+		// if the passed arg is just dir not followed by /...
+		if !recursiveLookup {
+			paths = append(paths, p)
+			continue
+		}
+
```

Comment:
Why not combine these checks?

`if !info.IsDir() || !recursiveLookup`

Reply:
Done: 7b84a173fcb791f3ec51fd46fcdb6f188312f4be